pilot/pkg/proxy/envoy/v2/ads_test.go

```diff
@@ -162,6 +165,393 @@ func TestAdsClusterUpdate(t *testing.T) {
+				index int
+				host  string
+			}{{index: 4}},
+			expectUpdates: []string{"cds"},
+		},
+		{
+			desc:            "Add a unscoped(name not match) service",
+			ev:              model.EventAdd,
+			svcNames:        []string{"foo.com"},
+			ns:              model.IstioDefaultConfigNamespace,
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		}, // then: default 1,2,3,4, foo.com; ns1: 11
+		{
+			desc: "Add instances to an unscoped service",
+			ev:   model.EventAdd,
+			instIndexes: []struct {
+				name    string
+				indexes []int
+			}{{"foo.com", []int{1, 2}}},
+			ns:              model.IstioDefaultConfigNamespace,
+			unexpectUpdates: []string{"eds"},
+			timeout:         time.Second,
+		}, // then: default 1,2,3,4
+		{
+			desc:            "Add a unscoped(ns not match) service",
+			ev:              model.EventAdd,
+			svcIndexes:      []int{11},
+			ns:              ns1,
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		}, // then: default 1,2,3,4, foo.com; ns1: 11
+		{
+			desc: "Add virtual service to an unscoped service",
+			ev:   model.EventAdd,
+			vsIndexes: []struct {
+				index int
+				hosts []string
+			}{{0, []string{"foo.com"}}},
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		},
+		{
+			desc: "Delete virtual service of a unscoped service",
+			ev:   model.EventDelete,
+			vsIndexes: []struct {
+				index int
+				hosts []string
+			}{{index: 0}},
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		},
+		{
+			desc: "Add destination rule to an unscoped service",
+			ev:   model.EventAdd,
+			drIndexes: []struct {
+				index int
+				host  string
+			}{{0, "foo.com"}},
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		},
+		{
+			desc: "Delete destination rule of a unscoped service",
+			ev:   model.EventDelete,
+			drIndexes: []struct {
+				index int
+				host  string
+			}{{index: 0}},
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		},
+		{
+			desc:          "Remove a scoped service",
+			ev:            model.EventDelete,
+			svcIndexes:    []int{4},
+			ns:            model.IstioDefaultConfigNamespace,
+			expectUpdates: []string{"lds"},
+		}, // then: default 1,2,3, foo.com; ns: 11
+		{
+			desc:            "Remove a unscoped(name not match) service",
+			ev:              model.EventDelete,
+			svcNames:        []string{"foo.com"},
+			ns:              model.IstioDefaultConfigNamespace,
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		}, // then: default 1,2,3; ns1: 11
+		{
+			desc:            "Remove a unscoped(ns not match) service",
+			ev:              model.EventDelete,
+			svcIndexes:      []int{11},
+			ns:              ns1,
+			unexpectUpdates: []string{"cds"},
+			timeout:         time.Second,
+		}, // then: default 1,2,3
+	}
+
+	for i, c := range svcCases {
+		fmt.Printf("begin %d case(%s) %v\n", i, c.desc, c)
```

Comment:
nit: `t.Run(c.desc, ...)`

That way it shows up independently if one case fails

Reply:
I've considered that. Changes will be pushed a little later.